src/riff-raff-yaml-file/index.ts

```diff
@@ -253,7 +254,18 @@ export class RiffRaffYamlFile {
             deployments.set(lambdaDeployment.name, lambdaDeployment.props);
           });
 
-          autoscalingGroups.forEach((asg) => {
+          // ASGs without an UpdatePolicy can be deployed via Riff-Raff's (legacy) `autoscaling` deployment type.
+          // ASGs with an UpdatePolicy are updated via Riff-Raff's `cloud-formation` deployment type.
+          const legacyAutoscalingGroups = autoscalingGroups.filter((asg) => {
+            const { cfnOptions } = asg.node.defaultChild as CfnAutoScalingGroup;
+            const { updatePolicy } = cfnOptions;
+            return (
+              updatePolicy?.autoScalingReplacingUpdate === undefined &&
```

Comment:
Minor: I think we've decided against using replacing update, so perhaps we can remove this line so it's clearer that we don't support this?

Reply:
Done.